met-web/src/components/engagement/form/EngagementWidgets/Documents/FileDrawer.tsx

```diff
@@ -53,13 +64,29 @@ const FileDrawer = () => {
         }
         try {
             setIsCreatingDocument(true);
-            await postDocument(widget.id, {
-                title: data.name,
-                parent_document_id: data.folderId === 0 ? null : data.folderId,
-                url: data.link,
-                widget_id: widget.id,
-                type: 'file',
-            });
+            if (documentToEdit) {
+                const documentEditsToPatch = updatedDiff(documentToEdit, {
```

Comment:
this is fine, perhaps do this though just to improve readability:

like in EngagementForm.tsx
    const handleSaveEngagement = () => {
        if (isNewEngagement) {
            return handleCreateEngagement();
        }

        return handleUpdateEngagement();
    };